fieldservice_sale/models/sale_order.py

```diff
@@ -41,25 +39,25 @@ def onchange_partner_id(self):
         they are FS locations.
         """
         res = super(SaleOrder, self).onchange_partner_id()
-        domain = [
-            "|",
-            "|",
-            ("partner_id", "=", self.partner_id.id),
-            ("partner_id", "=", self.partner_shipping_id.id),
-            ("partner_id", "=", self.partner_id.commercial_partner_id.id),
-        ]
-        if self.partner_id.fsm_location:
-            domain = [("partner_id", "=", self.partner_id.id)]
-        location_ids = self.env["fsm.location"].search(domain)
-        self.fsm_location_id = location_ids and location_ids[0] or False
+        location = self.partner_id.fsm_location
+        if not location:
+            domain = [
+                "|",
+                "|",
+                ("partner_id", "=", self.partner_id.id),
+                ("partner_id", "=", self.partner_shipping_id.id),
+                ("partner_id", "=", self.partner_id.commercial_partner_id.id),
+            ]
```

Comment:
can you pls use OR expression from osv?
```
expression.OR(
    [
        [("partner_id", "=", self.partner_id.id)],
        [("partner_id", "=", self.partner_shipping_id.id)],
        [("partner_id", "=", self.partner_id.commercial_partner_id.id)],
    ]
)
```

Reply:
Why? We're not composing domain dinamically, so we know beforehand which conditions are to be chained with an `OR`, so we can create the domain directly using the `"|"` operator. It's easier to read, and doesn't force Odoo to compute a new domain.